mbtools/copy_resources_s3.py

```diff
@@ -0,0 +1,137 @@
+import hashlib
+import json
+import os
+import sys
+import magic
+import boto3
+import botocore
+from pathlib import Path
+
+
+def upload_resources(resource_dir, metadata_dir, bucket, s3_dir):
+    """Uplaods resources to s3 if they dont already exist there"""
+    hash_to_filename_map = new_resource_hashes(resource_dir)
+    hashes_new = list(hash_to_filename_map.keys())
+    hashes_metadata = existing_metadata_hashes(metadata_dir)
+    hashes_s3 = existing_s3_hashes(bucket, s3_dir)
+    hashes_new = compare_and_remove_hashes(hashes_new, hashes_metadata)
+    hashes_new = compare_and_remove_hashes(hashes_new, hashes_s3)
+    add_new_resources_to_s3(bucket, s3_dir, hashes_new, hash_to_filename_map,
+                            metadata_dir
+                            )
+
+
+def existing_metadata_hashes(dir):
+    """Gathers the sha1s for existing resources from a local metadata json"""
+    hashes = []
+    try:
+        with open(dir) as f:
+            data = json.load(f)
+            for item in data:
+                hashes.append(item["sha1"])
+        return hashes
+    except FileNotFoundError:
+        return FileNotFoundError
```

Comment:
Yeah, I would just let the exception flow and not incorporate a `try / except` since you're not doing any failure handling here. 

Though, give you have `Path(sys.argv[2]).resolve(strict=True)` in `main()`, I think the `FileNotFoundError` would get raised there if the metadata file input is incorrect.